tests/framework/db/QueryTest.php

```diff
@@ -110,6 +110,60 @@ public function testFilterWhere()
         $this->assertEquals($condition, $query->where);
     }
 
+    public function testFilterHaving()
+    {
+        // should work with hash format
+        $query = new Query;
+        $query->filterHaving([
+            'id' => 0,
+            'title' => '   ',
+            'author_ids' => [],
+        ]);
+        $this->assertEquals(['id' => 0], $query->having);
+
+        $query->andFilterHaving(['status' => null]);
+        $this->assertEquals(['id' => 0], $query->having);
+
+        $query->orFilterHaving(['name' => '']);
+        $this->assertEquals(['id' => 0], $query->having);
+
+        // should work with operator format
+        $query = new Query;
+        $condition = ['like', 'name', 'Alex'];
+        $query->filterHaving($condition);
+        $this->assertEquals($condition, $query->having);
+
+        $query->andFilterHaving(['between', 'id', null, null]);
+        $this->assertEquals($condition, $query->having);
+
+        $query->orFilterHaving(['not between', 'id', null, null]);
+        $this->assertEquals($condition, $query->having);
+
+        $query->andFilterHaving(['in', 'id', []]);
+        $this->assertEquals($condition, $query->having);
+
+        $query->andFilterHaving(['not in', 'id', []]);
+        $this->assertEquals($condition, $query->having);
+
+        $query->andFilterHaving(['not in', 'id', []]);
```

Comment:
this one is the same as the previous. If thats on purpose, please add a comment description